.github/workflows/main.yml

```diff
@@ -55,6 +55,10 @@ jobs:
         run: |
           set -euo pipefail
           go test -p 1 -json ./... -covermode=count -coverprofile=coverage.out 2>&1 | tee test_results.json
+      - name: Sanitize test results
+        # We're utilizing `tee` above which can capture non-json stdout output, we also need to remove non-json lines.
+        if: ${{ always() && env.TARGET_GOLANG_VERSION == matrix.go }}
+        run: cat test_results.json | jq -c -R 'fromjson? | select(type == "object")' > tmp.json && mv tmp.json test_results.json
```

Comment:
Curious if this works if we do `> test_results.json` instead of `> tmp.json && mv tmp.json test_results.json`?

Reply:
@Olshansk I didn't want to `cat` and forward `stdout` to the same file at the same time. To be frank, I don't even know if this will work. `jq` also didn't provide a way to put changes in place, so I resorted to output to the temporary file instead.

Reply:
`To be frank, I don't even know if this will work`

You're not sure if `fromjson` will work?

Reply:
@Olshansk `fromjson?` works, and I've validated it.

Your question was if `> test_results.json` could work instead of `> tmp.json && mv tmp.json test_results.json`. My answer is I don't know, but I wouldn't do it anyway—because we're doing `cat` against the same file: `cat test_results.json > test_results.json`, just additionally processing via pipe `| jq -c -R 'fromjson? | select(type == "object")'`.